codalab/lib/download_manager.py

```diff
@@ -66,7 +81,7 @@ def get_target_info(self, uuid, path, depth):
             try:
                 return download_util.get_target_info(bundle_path, uuid, path, depth)
             except download_util.PathException as e:
-                raise NotFoundError(str(e))
+                return check_dependencies(NotFoundError(str(e)))
```

Comment:
I think this logic is a bit non-obvious, so documenting would be good.  Also, I wonder if the recursive structure here could be streamlined...I think it could be cleaner to have a separate helper function that shouldn't include the handling of other bundle states.

Reply:
I'll take an attempt at this.